lib/sgr.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+/* CSI - control sequence introducer */
+/* SGR - set graphic rendition */
+
+var assign   = require('es5-ext/object/assign')
+  , includes = require('es5-ext/string/#/contains');
+
+var CSI = '\x1b[';
+
+var sgr = function (code) {
+	return CSI + code + 'm';
+};
+
+sgr.CSI = CSI;
+
+var mods = assign({
+	// Style
+	bold:      { _bold: [1, 22] },
+	italic:    { _italic: [3, 23] },
+	underline: { _underline: [4, 24] },
+	blink:     { _blink: [5, 25] },
+	inverse:   { _inverse: [7, 27] },
+	strike:    { _strike: [9, 29] }
+
+	// Color
+}, ['black', 'red', 'green', 'yellow', 'blue', 'magenta', 'cyan', 'white']
+	.reduce(function (obj, color, index) {
+		// foreground
+		obj[color] = { _fg: [30 + index, 39] };
+		obj[color + 'Bright'] = { _fg: [90 + index, 39] };
+
+		// background
+		obj['bg' + color[0].toUpperCase() + color.slice(1)] = { _bg: [40 + index, 49] };
+		obj['bg' + color[0].toUpperCase() + color.slice(1) + 'Bright'] = { _bg: [100 + index, 49] };
+
+		return obj;
+	}, {}));
+
+sgr.mods = mods;
+
+(function () {
+	var forEach = require('es5-ext/object/for-each')
+	  , onlyKey = require('es5-ext/object/first-key')
+	  , uniq = require('es5-ext/array/#/uniq.js');
+
+	sgr.toModPair = function (mod) {
+		return mod[onlyKey(mod)];
+	};
+
+	var openers = {}, closers = {};
+
+	forEach(mods, function (mod) {
+		var modPair = sgr.toModPair(mod);
```

Comment:
`key` here is function name (`bold`, `magenta`, `red` etc), while we need `fg`, `bg`-stuff.
```js
console.log(key, onlyKey(mod));
// strike _strike
// black _fg
// blackBright _fg
// …
```
But I've removed `toModPair`, since it've been used once in code.